sql/src/main/java/io/crate/executor/transport/AlterTableOperation.java

```diff
@@ -107,7 +108,58 @@ public AlterTableOperation(ClusterService clusterService,
         return result;
     }
 
-    private class ResultSetReceiver implements ResultReceiver {
+    public CompletableFuture<Long> executeAlterTableOpenClose(final AlterTableOpenCloseAnalyzedStatement analysis) {
+        List<CompletableFuture<Long>> results = new ArrayList<>(2);
```

Comment:
first check should be to check if any operation is needed at all (close closed table, open opened table). if no operation is needed, we can return an immediate future